proxy.go

```diff
@@ -107,6 +107,8 @@ func (s *validTailnetSrv) rewrite(r *httputil.ProxyRequest) {
 			r.Out.Header.Set("X-Forwarded-Server", hostOnly)
 			r.Out.Header.Set("X-Forwarded-Port", port)
 		}
+    r.Out.Header.Set("Upgrade", r.In.Header.Get("Upgrade"))
```

Comment:
I'd love to add a commandline switch for this - that go's ReverseProxy strips the header out by default does worry me a bit about upstream servers that might not expect upgrade headers being set at all.